src/redux/reducers.js

```diff
@@ -1154,7 +1154,8 @@ export function user(state = initUser(), action) {
     case response(ActionTypes.UPDATE_USER):
     case response(ActionTypes.UPDATE_FRONTEND_PREFERENCES):
     case response(ActionTypes.UPDATE_FRONTEND_REALTIME_PREFERENCES): {
-      return {...state, ...userParser(action.payload.users)};
+      const subscriptions = _.uniq((action.payload.subscriptions || []).map(sub => sub.user));
```

Comment:
I can move WHO_AM_I and SIGN_UP into the switch, but there's still gonna be repeated piece because of `isUserChangeResponse`.